k8s/k8s.go

```diff
@@ -172,23 +172,36 @@ func CmdAddK8s(args *skel.CmdArgs, conf utils.NetConf, hostname string, calicoCl
 		}
 
 		ipAddrsNoIpam := annot["cni.projectcalico.org/ipAddrsNoIpam"]
-
-		// Call IPAM plugin if ipAddrsNoIpam annotation is not present.
-		if ipAddrsNoIpam == "" {
-			// Run the IPAM plugin
+		ipAddrs := annot["cni.projectcalico.org/ipAddrs"]
+
+		// switch based on which annotations are passed or not passed.
+		switch {
+		// Can't have both ipAddrs and ipAddrsNoIpam annotations at the same time.
```

Comment:
Be consistent with the placement of your comments in the switch statement.  My preference is for the comments to go within each case.